llama_stack/core/conversations/conversations.py

```diff
@@ -0,0 +1,290 @@
+        """Get a conversation with the given ID."""
+        record = await self.sql_store.fetch_one(table="openai_conversations", where={"id": conversation_id})
+
+        if record is None:
+            raise ValueError(f"Conversation {conversation_id} not found")
+
+        return Conversation(
+            id=record["id"], created_at=record["created_at"], metadata=record.get("metadata"), object="conversation"
+        )
+
+    async def update_conversation(self, conversation_id: str, metadata: Metadata) -> Conversation:
+        """Update a conversation's metadata with the given ID"""
+        await self.sql_store.update(
+            table="openai_conversations", data={"metadata": metadata}, where={"id": conversation_id}
+        )
+
+        return await self.get_conversation(conversation_id)
+
+    async def openai_delete_conversation(self, conversation_id: str) -> ConversationDeletedResource:
+        """Delete a conversation with the given ID."""
+        await self.sql_store.delete(table="openai_conversations", where={"id": conversation_id})
+
+        logger.info(f"Deleted conversation {conversation_id}")
+        return ConversationDeletedResource(id=conversation_id)
+
+    def _validate_conversation_id(self, conversation_id: str) -> None:
+        """Validate conversation ID format."""
+        if not conversation_id.startswith("conv_"):
+            raise ValueError(
+                f"Invalid 'conversation_id': '{conversation_id}'. Expected an ID that begins with 'conv_'."
+            )
+
+    async def _get_validated_conversation(self, conversation_id: str) -> Conversation:
+        """Validate conversation ID and return the conversation if it exists."""
+        self._validate_conversation_id(conversation_id)
+        return await self.get_conversation(conversation_id)
+
+    async def create(self, conversation_id: str, items: list[ConversationItem]) -> ConversationItemList:
+        """Create items in the conversation."""
+        await self._get_validated_conversation(conversation_id)
+
+        created_items = []
+
+        for item in items:
+            # Generate item ID based on item type
+            random_bytes = secrets.token_bytes(24)
+            item_type = getattr(item, "type", None)
+            if item_type == "message":
+                item_id = f"msg_{random_bytes.hex()}"
+            else:
+                item_id = f"item_{random_bytes.hex()}"
+
+            # Create a copy of the item with the generated ID and completed status
+            item_dict = item.model_dump() if hasattr(item, "model_dump") else dict(item)
+            item_dict["id"] = item_id
+            if "status" not in item_dict:
+                item_dict["status"] = "completed"
+
+            created_items.append(item_dict)
+
+        # Get existing items from database
+        record = await self.sql_store.fetch_one(table="openai_conversations", where={"id": conversation_id})
```

Comment:
this seems like the wrong semantic. I don't think the API ever says one is appending. a create is a create.